tests/common/test_event.py

```diff
@@ -437,107 +439,214 @@ def test_elapsed_milliseconds(self):
+
+    def test_collect_events_should_use_the_file_creation_time_for_legacy_agent_events_missing_a_timestamp(self):
+        test_file = self._create_test_event_file("legacy_agent_no_timestamp.tld")
+
+        event_creation_time = TestEvent._get_file_creation_timestamp(test_file)
+
+        event_list = event.collect_events()
+
+        self.assertEquals(len(event_list.events), 1)
+
+        self._assert_event_includes_all_parameters_in_the_telemetry_schema(
+            event_list.events[0],
+            expected_parameters={
+                "Name": "WALinuxAgent",
+                "Version": "9.9.9",
+                "IsInternal": False,
+                "Operation": "InitializeCGroups",
+                "OperationSuccess": True,
+                "Message": "The cgroup filesystem is ready to use",
+                "Duration": 1234,
+                "ExtensionType": "ALegacyExtensionType",
+                "GAVersion": "WALinuxAgent-1.1.1",
+                "ContainerId": "AAAAAAAA-BBBB-CCCC-DDDD-EEEEEEEEEEEE",
+                "EventTid": 98765,
+                "EventPid": 4321,
+                "TaskName": "ALegacyTask",
+                "KeywordName": "ALegacyKeywordName"},
+            assert_timestamp=lambda timestamp:
+                self.assertEquals(timestamp, event_creation_time, "The event timestamp (opcode) is incorrect")
+        )
+
+    def _assert_extension_event_includes_all_parameters_in_the_telemetry_schema(self, event_file):
+        # Extensions drop their events as *.tld files on the events directory. They populate only a subset of fields,
+        # and the rest are added by the agent when events are collected.
+        test_file = self._create_test_event_file(event_file)
+
+        event_creation_time = TestEvent._get_file_creation_timestamp(test_file)
+
+        event_list = event.collect_events()
+
+        self.assertEquals(len(event_list.events), 1)
+
+        self._assert_event_includes_all_parameters_in_the_telemetry_schema(
+            event_list.events[0],
+            expected_parameters={
+                'Name': 'Microsoft.Azure.Extensions.CustomScript',
+                'Version': '2.0.4',
+                'Operation': 'Scenario',
+                'OperationSuccess': True,
+                'Message': 'A test telemetry message.',
+                'Duration': 150000,
+                'ExtensionType': 'json'},
+            assert_timestamp=lambda timestamp:
+                self.assertEquals(timestamp, event_creation_time, "The event timestamp (opcode) is incorrect")
+            )
+
+    def test_collect_events_should_add_all_the_parameters_in_the_telemetry_schema_to_extension_events(self):
+        self._assert_extension_event_includes_all_parameters_in_the_telemetry_schema('custom_script.tld')
+
+    def test_collect_events_should_ignore_extra_parameters_in_extension_events(self):
+        self._assert_extension_event_includes_all_parameters_in_the_telemetry_schema('custom_script_extra_parameters.tld')
```

Comment:
We should also test extension events that are XML, not just JSON. The original need for this PR was because we were incorrectly handling XML extension events, because they already had `ContainerId` present, but empty, which we honored instead of overriding the value.

There is an existing (real) VMAccess XML event in `\tests\data\ext\event_from_extension.xml`.

Reply:
The filename `event_from_extension.xml` should be updated too, since extension events are `.tld`, regardless of if they're XML or JSON.

Reply:
Discussed offline - this is covered in the test case using custom_script_extra_parameters.tld